api/data_handler.go

```diff
@@ -64,11 +67,16 @@ func (handler *DataHandler) PostData(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err = handler.memStore.HandleIngestion(postDataRequest.TableName, postDataRequest.Shard, upsertBatch)
-	if err != nil {
-		common.RespondWithError(w, err)
-		return
-	}
+	available := handler.workerPool.GoIfAvailable(func() {
```

Comment:
for ingestion, better do throttling instead of error out, or need to change the client to wait/retry

Reply:
client already does retry